servers/src/main/java/tachyon/master/file/meta/Inode.java

```diff
@@ -156,10 +167,28 @@ public boolean isFile() {
   }
 
   /**
-   * Reverse the id of the inode. Only used for a delete operation.
+   * Get the pinned flag of the inode
+   *
+   * @return true if the inode is pinned, false otherwise
+   */
+  public synchronized boolean isPinned() {
```

Comment:
the pin status is propagated when creating new inodes. the pin status should probably be in both dirs and files in order to have access to the pin information.